infrastructure/modules/ECR/main.tf

```diff
@@ -0,0 +1,28 @@
+module "ecr" {
+  source = "terraform-aws-modules/ecr/aws"
+
+  repository_name = "private-ecr-repo"
+
```

Comment:
You are missing **repository_read_write_access_arns**

Reply:
Value should be dynamically set from the iam role that you have made in different module.
P.s. Please don't resolve conversations, I will resolve them once I confirm that change is appropriate.

Reply:
Isn't it dynamically created? I took the iam role from iam output file and then used it in root main file where I assign it to a variable. Then variable is assign to the repository_read_write_access_arns in ECR file.

Reply:
Variable part should be dynamic, this one:

**var.repository_read_write_access_arns**

Instead you should call for module to look for the arn output of module that created iam role itself.